File: LivrusVirtus/VCPP6/FoxToolKit/ADO/ConEditoras.cpp
#include "resource.h"
#include "ConEditoras.h"
#include "Principal.h"
#include "ConsultasSQL.h"

CfrmConEditoras *frmConEditoras = (CfrmConEditoras *) NULL;

FXDEFMAP(CfrmConEditoras) CfrmConEditorasMap[]={
  FXMAPFUNC(SEL_COMMAND, CfrmConEditoras::ID_FRMCONSULTAS_BTNPESQUISAR, 
    CfrmConEditoras::onBtnPesquisar),
  FXMAPFUNC(SEL_COMMAND, CfrmConEditoras::ID_FRMCONSULTAS_BTNLIMPAR, 
    CfrmConEditoras::onBtnLimpar),
  FXMAPFUNC(SEL_SELECTED,CfrmConEditoras::ID_FRMCONSULTAS_GRIDCONSULTA,
    CfrmConEditoras::onGridSelected),
  FXMAPFUNC(SEL_CHANGED,CfrmConEditoras::ID_FRMCONSULTAS_GRIDCONSULTA,
    CfrmConEditoras::onGridChanged)
};
FXIMPLEMENT(CfrmConEditoras,CfrmConsultas,CfrmConEditorasMap,
  ARRAYNUMBER(CfrmConEditorasMap))

CfrmConEditoras::CfrmConEditoras(FXWindow* owner):
  CfrmConsultas(owner, STR_CONEDITORA) {
  IniciaControles();
}

void CfrmConEditoras::IniciaControles() {
  this->labDescricao->setText("Editora:");
  this->gridConsulta->setColumnText(1, "Editora");
}

void CfrmConEditoras::create(){
  FXDialogBox::create();
  show(PLACEMENT_SCREEN);
}

CfrmConEditoras::~CfrmConEditoras(){  
}

long CfrmConEditoras::onBtnPesquisar(FXObject*,FXSelector,void*) {
  PesquisarDados(
    ConsSQL->Editora('S',"", edtDescricao->getText()), STR_EDTNENC, 2);
  return true;
}

long CfrmConEditoras::onBtnLimpar(FXObject*,FXSelector,void*) {
  LimparDados();
  return true;
}

long CfrmConEditoras::onGridSelected(FXObject*,FXSelector,void* ptr){
  FXTableRange *escala = (FXTableRange*) ptr;
  setLabRegistros(escala->fm.row);
  return true;
}

long CfrmConEditoras::onGridChanged(FXObject*,FXSelector,void* ptr){
  FXTablePos *pos = (FXTablePos*) ptr;
  setLabRegistros(pos->row);
  return true;
}

File: LivrusVirtus/FoxToolKit/FXPython/ConEditoras.py
from FXPy.fox import *
from Consultas import frmConsultas
from resource import Resource
from ConsultasSQL import ConsultasSQL
import Principal

class frmConEditoras(frmConsultas):

  def __init__(self, owner):
    frmConsultas.__init__(self, owner, Resource.STR_CONEDITORA)  

    FXMAPFUNC(self, SEL_COMMAND, frmConEditoras.ID_FRMCONSULTAS_BTNPESQUISAR, 
      frmConEditoras.onBtnPesquisar)
    FXMAPFUNC(self, SEL_COMMAND, frmConEditoras.ID_FRMCONSULTAS_BTNLIMPAR, 
      frmConEditoras.onBtnLimpar)
    FXMAPFUNC(self, SEL_SELECTED, frmConEditoras.ID_FRMCONSULTAS_GRIDCONSULTA, 
      frmConEditoras.onGridSelected)
    FXMAPFUNC(self, SEL_CHANGED, frmConEditoras.ID_FRMCONSULTAS_GRIDCONSULTA, 
      frmConEditoras.onGridChanged)
    
    self.labDescricao.setText("Editora:")
    self.gridConsulta.setItemText(0, 2, "Editora")

  def create(self):
    frmConsultas.create(self)
  
  def onBtnPesquisar(self, sender, sel, ptr):
    self.PesquisarDados(ConsultasSQL.ConsSQL.Editora('S',"", 
      self.edtDescricao.getText()), Resource.STR_EDTNENC, 2)
    return True
  
  def onBtnLimpar(self, sender, sel, ptr):
    self.LimparDados()
    return True
  
  def onGridSelected(self, sender, sel, ptr):
    self.setLabRegistros(sender.getCurrentRow())
    return True
  
  def onGridChanged(self, sender, sel, ptr):
    self.setLabRegistros(sender.getCurrentRow())
    return True
